### PR TITLE
[cmd/builder] Add option to skip generating sources

### PR DESCRIPTION
**Pull request by hypnoce**
This option makes the builder skip generating sources.

**Link to tracking Issue:** #7541

**Testing:** Unit tests

**Documentation:** as part of the CLI help menu

### CODE REVIEW

File: cmd/builder/internal/command.go
Comment (bogdandrutu) [inline cmd/builder/internal/command.go:89]: If it skips generating go code, what does it do?

Comment (hypnoce) [reply cmd/builder/internal/command.go:89]: Depending on the other options (SkipCompilation and SkipGetModules), it can fetch modules and or build.

Comment (bogdandrutu) [reply cmd/builder/internal/command.go:89]: But that is a "go" command. If anything this tool should not build at all.

Comment (hypnoce) [reply cmd/builder/internal/command.go:89]: You suggest to remove building from the builder ?
Besides, do you know why it's possible to skip module sync but still compiling ? I guess that's also an invalid use case since the go.mod file is not in sync with the generated code in that case.

**Review by jpkrohling — COMMENTED** [review]
This change deserves some good documentation as part of the readme, especially around the target use case. Other than that, it looks good to me, but I'll let the maintainers judge whether this is a feature worth having.

**Comment by hypnoce** [comment]
> This change deserves some good documentation as part of the readme, especially around the target use case. Other than that, it looks good to me, but I'll let the maintainers judge whether this is a feature worth having.

I've updated the README with clear use cases. Let me know if that's clear enough.

**Review by jpkrohling — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Comment by hypnoce** [comment]
@jpkrohling I've added a unit test for SkipGenerate. Let me know what else I can do. Thanks

File: cmd/builder/README.md
Comment (jpkrohling) [inline cmd/builder/README.md:118]:
```suggestion
The builder has 3 steps:
```

File: cmd/builder/README.md
Comment (jpkrohling) [inline cmd/builder/README.md:122]:
```suggestion
* Compilation: builds the OpenTelemetry Collector executable
```

**Review by jpkrohling — COMMENTED** [review]
This looks good, apart from a couple of typos, but we still need @bogdandrutu's blessing, given he had concerns previously about this.

**Review by codeboten — APPROVED** [review]
It's a small enough change, it solves a use-case, and it's the only step that is currently not skippable. I'm ok w/ this change.

**Pull request merged**
